src/daft-sql/src/functions.rs

```diff
@@ -382,8 +382,139 @@ impl SQLPlanner<'_> {
+                        window_spec.order_by.push(parsed_expr);
+                        window_spec
+                            .descending
+                            .push(order.asc.is_some_and(|asc| !asc));
+                    }
+                }
+
+                if spec.window_frame.is_some() {
+                    if let Some(sql_frame) = &spec.window_frame {
+                        let frame_type = match sql_frame.units {
+                            sqlparser::ast::WindowFrameUnits::Rows => WindowFrameType::Rows,
+                            sqlparser::ast::WindowFrameUnits::Range => WindowFrameType::Range,
+                            sqlparser::ast::WindowFrameUnits::Groups => {
+                                unsupported_sql_err!("Window frame units: Groups")
+                            }
+                        };
+
+                        let start = self.convert_window_frame_bound(&sql_frame.start_bound)?;
+
+                        // Convert end bound or default to CURRENT ROW if not specified
+                        let end = match &sql_frame.end_bound {
+                            Some(end_bound) => self.convert_window_frame_bound(end_bound)?,
+                            None => WindowBoundary::Offset(0), // CURRENT ROW
+                        };
+
+                        window_spec.frame = Some(WindowFrame {
+                            frame_type,
+                            start,
+                            end,
+                        });
+                    }
+                }
+
+                // TODO: This should probably be done later in the code
+                if let Some(current_plan) = &self.current_plan {
+                    window_spec = current_plan.resolve_window_spec(window_spec)?;
+                }
+
+                Ok(window_spec)
+            }
+            WindowType::NamedWindow(_) => {
+                unsupported_sql_err!("Named windows are not supported yet")
+            }
+        }
+    }
+
+    /// Helper function to convert SQLParser's WindowFrameBound to Daft's WindowBoundary
+    fn convert_window_frame_bound(
+        &self,
+        bound: &sqlparser::ast::WindowFrameBound,
+    ) -> SQLPlannerResult<daft_dsl::expr::window::WindowBoundary> {
+        use daft_dsl::expr::window::WindowBoundary;
+
+        match bound {
+            sqlparser::ast::WindowFrameBound::CurrentRow => Ok(WindowBoundary::Offset(0)),
+            sqlparser::ast::WindowFrameBound::Preceding(None) => {
+                Ok(WindowBoundary::UnboundedPreceding())
+            }
+            sqlparser::ast::WindowFrameBound::Following(None) => {
+                Ok(WindowBoundary::UnboundedFollowing())
+            }
+            sqlparser::ast::WindowFrameBound::Preceding(Some(expr)) => {
+                let parsed_expr = self.plan_expr(expr)?;
+
+                if let Some(lit) = parsed_expr.as_literal() {
+                    if let Some(value) = lit.as_i64() {
```

Comment:
nit: we can chain here so we don't need to do nested `if let Some`'s 

```suggestion
                if let Some(value) = parsed_expr.as_literal().and_then(|lit| lit.as_i64()) {

```